sk0-back/src/users/services/users/users.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@nestjs/common';
+import { CreateUserDto } from 'src/users/dtos/createUser.dto';
+import { User } from 'src/users/types/User';
+
+@Injectable()
+export class UsersService {
+  private users: User[] = [
+    {
+      id: 2,
+      username: 'testuser',
+      email: 'testuser@example.com',
+      password: 'password1234',
+    },
+  ];
+
+  createUser(userDto: CreateUserDto) {
+    this.users.push(userDto);
```

Comment:
Szerintem valami ilyesmi kellene bele:
```suggestion
    const usersExist = getByEmail(userDto.email);
    if (usersExist) {
      throw(new ConflictException)
    }
    this.users.push(userDto);
```

Reply:
a controllerben van ilyesmi. inkabb a serviceben legyen? vagy mindkettoben?

Reply:
Ez benne van már a userservicemben. ezt egyébként érdemes a serviceben kezelni mindig. a controller általában a végpontok requestjeit irányítja. Az üzleti logikát pedig kiszervezzük servicekbe. @markkovari  Kérlek erősíts vagy cáfolj meg, nem akarok butaságot mondani.

Reply:
Exactly.